Tests/FluentAssertions.Equivalency.Specs/BasicSpecs.cs

```diff
@@ -645,4 +645,23 @@ public void When_the_graph_contains_guids_it_should_properly_format_them()
         // Assert
         act.Should().Throw<XunitException>().WithMessage("Expected property actual[0].Id*to be *-*, but found *-*");
     }
+
+    [Fact]
+    public void When_a_class_contains_an_empty_array_segment()
+    {
+        // Arrange
+        var actual = new ClassWithArraySegment();
+        var expected = new ClassWithArraySegment();
+
+        // Act
+        Action act = () => actual.Should().BeEquivalentTo(expected);
+
+        // Assert
+        act.Should().NotThrow();
```

Comment:
```suggestion
        // Act / Assert
        actual.Should().BeEquivalentTo(expected);
```

Just call the assertion. If it fails, it fails anyway.

Reply:
Addressed in https://github.com/fluentassertions/fluentassertions/pull/2511